File: core/src/engine/rts/model/TroopDef.scala
package engine.rts.model

import server.model.playerData.Player

object TroopDef{
    val DEAD_STATE="dead"
    val FREEZE_STATE="noMove"
    val NORMAL_STATE="canMove"
}

class TroopDef(val owner:Player) {
    var myState=TroopDef.NORMAL_STATE
    def die()=myState=TroopDef.DEAD_STATE
    def canMove()=(myState==TroopDef.NORMAL_STATE)
}

File: core/src/engine/rts/model/TroopDef.java
package engine.rts.model;

import server.model.playerData.Player;

public class TroopDef{
    
	static final String DEAD_STATE="dead";
    static final String FREEZE_STATE="noMove";
    static final String NORMAL_STATE="canMove";
    
    final Player owner;
    private String myState=TroopDef.NORMAL_STATE;
    
    public TroopDef(Player o){
    	owner=o;
    }
    
    public void die(){
    	myState=TroopDef.DEAD_STATE;
    }
    
    public boolean canMove(){
    	return myState==TroopDef.NORMAL_STATE;
    }
    
   
}